### PR TITLE
[Enhancement] Refactor parquet schema parse logic and improve stability

### PR DESCRIPTION
**Pull request by Smith-Cruise**
This this pr, we do the following things:
1. Add more defensive code, to avoid BE crashed
2. Return false in RLE decoder instead of DCHECK(false)
3. Rewrite the schema parse code, improved code readability, and remove some illegal parquet file support. The code borrows from the implementation of ARROW, but the general logic remains unchanged. ARROW schema parse logic: https://github.com/apache/arrow/blob/main/cpp/src/parquet/arrow/schema.cc
4. Borrow the schema ut from ARROW: https://github.com/apache/arrow/blob/main/cpp/src/parquet/arrow/arrow_schema_test.cc
5. **In the previous version, we will parse `MAP<INT, INT>` into `MAP<STRUCT<KEY, VALUE>>`, now we fixed it.**
6. Don't support some illegal parquet files anymore, these files are reported by QA's HUDI cluster, and in the newest HUDI version, these bugs are already fixed. These bug parquet files can be read by Spark only.
7. Write parquet_cli tools, it can read parquet files directly, and we don't need to set schema information in code anymore.
8. Add [arrow's fuzzing test files](https://github.com/apache/arrow-testing/tree/master/data/parquet).

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/formats/parquet/parquet_cli_reader.h [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+#include "common/status.h"
+#include "formats/parquet/file_reader.h"
+#include "fs/fs.h"
+#include "runtime/descriptor_helper.h"
+
+namespace starrocks::parquet {
+
+class ParquetCLIReader {
+public:
+    ParquetCLIReader(const std::string& filepath)
```
Comment (letian-jiang) [inline be/src/formats/parquet/parquet_cli_reader.h:29]: What is this class for? Unit testing?

Comment (Smith-Cruise) [reply be/src/formats/parquet/parquet_cli_reader.h:29]: Yes, of course we can implement an parquet cli reader later, just like `parquet-tools`

File: be/src/formats/parquet/schema.h [hunk 2]
```diff
@@ -89,30 +131,59 @@ class SchemaDescriptor {
...
 private:
-    void leaf_to_field(const tparquet::SchemaElement& t_schema, const LevelInfo& cur_level_info, bool is_nullable,
-                       ParquetField* field);
+    //    void leaf_to_field(const tparquet::SchemaElement& t_schema, const LevelInfo& cur_level_info, bool is_nullable,
+    //                       ParquetField* field);
+
+    Status leaf_to_field(const tparquet::SchemaElement* t_schema, const LevelInfo& cur_level_info, bool is_nullable,
+                         ParquetField* field);
+
+    //    Status list_to_field(const std::vector<tparquet::SchemaElement>& t_schemas, size_t pos, LevelInfo cur_level_info,
```
Comment (letian-jiang) [inline be/src/formats/parquet/schema.h:147]: duplicate comments?

File: be/src/formats/parquet/schema.cpp [hunk 3]
```diff
@@ -46,49 +46,92 @@ std::string ParquetField::debug_string() const {
...
-static bool is_group(const tparquet::SchemaElement& schema) {
-    return schema.num_children > 0;
-}
+//static bool is_group(const tparquet::SchemaElement& schema) {
+//    return schema.num_children > 0;
+//}
 
-static bool is_list(const tparquet::SchemaElement& schema) {
-    return schema.__isset.converted_type && schema.converted_type == tparquet::ConvertedType::LIST;
+static bool is_group(const tparquet::SchemaElement* schema) {
```
Comment (letian-jiang) [inline be/src/formats/parquet/schema.cpp:53]: what's the point of changing reference to pointer?

Comment (Smith-Cruise) [reply be/src/formats/parquet/schema.cpp:53]: In order to return `StatusOr<SchemaElement*>`

File: be/src/formats/parquet/parquet_cli_reader.h
Comment (DorianZheng) [inline be/src/formats/parquet/parquet_cli_reader.h:1]: If this file is only used by UT, please move it to test directory.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by letian-jiang — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/formats/parquet/schema.h [hunk 4]
```diff
@@ -19,27 +19,69 @@
...
+    // has the def levels [0, 1, 2, 3, 4].  The actual
+    // struct array is only of length 3: [not-set, set, set] and
+    // the int array is also of length 3: [N/A, null, 1].
+    //
     int16_t immediate_repeated_ancestor_def_level = 0;
 
-    bool is_nullable() const { return max_def_level > immediate_repeated_ancestor_def_level; }
-
     int16_t increment_repeated() {
         auto origin_ancestor_rep_levels = immediate_repeated_ancestor_def_level;
```
Comment (zombee0) [inline be/src/formats/parquet/schema.h:67]: could you rename origin_ancestor_rep_levels to  origin_ancestor_def_levels? it's on def levels

**Review by zombee0 — APPROVED** [review]

**Comment by Smith-Cruise** [comment]
@zombee0 PTAL

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/formats/parquet/schema.h (hunk 4 above)
Comment (Smith-Cruise) [reply be/src/formats/parquet/schema.h]: fixed

File: be/src/util/bit_stream_utils.inline.h [hunk 5]
```diff
@@ -205,18 +205,38 @@ inline bool BitReader::GetAligned(int num_bytes, T* v) {
...
+//    uint8_t byte = 0;
+//    do {
+//        if (!GetAligned<uint8_t>(1, &byte)) return false;
+//        *v |= (byte & 0x7F) << shift;
+//        shift += 7;
+//        DCHECK_LE(++num_bytes, MAX_VLQ_BYTE_LEN);
+//    } while ((byte & 0x80) != 0);
+//    return true;
+//}
+
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply be/src/util/bit_stream_utils.inline.h]: fixed

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by dirtysalt — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**